cdap-common/src/main/java/co/cask/cdap/common/service/UncaughtExceptionIdleService.java

```diff
@@ -17,22 +17,26 @@
 package co.cask.cdap.common.service;
 
 import com.google.common.util.concurrent.AbstractIdleService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.util.concurrent.Executor;
 
 /**
  * An AbstractIdleService that overrides its executor to use one that doesn't print to stderr the exceptions
- * it gets, so that error messaging is left to whoever is using the service.
+ * it gets, but instead
  */
-public abstract class UnloggedExceptionIdleService extends AbstractIdleService {
+public abstract class UncaughtExceptionIdleService extends AbstractIdleService {
   public static final Thread.UncaughtExceptionHandler UNCAUGHT_EXCEPTION_HANDLER =
     new Thread.UncaughtExceptionHandler() {
       @Override
       public void uncaughtException(Thread t, Throwable e) {
-        // no-op
+        LOG.error("Uncaught exception from " + t.toString(), e);
       }
     };
 
+  private static final Logger LOG = LoggerFactory.getLogger(UncaughtExceptionIdleService.class);
```

Comment:
this will make all the log messages come from this class rather than the actual service. Perhaps we can add an abstract method getUncaughtExceptionLogger() for subclasses to implement, and that executor() will use to create its UncaughtExceptionHandler